addons/common/fnc_getObjectConfig.sqf

```diff
@@ -19,24 +20,39 @@ Example:
     (end)
 
 Author:
-    commy2
+    commy2, kuloodporny
 ---------------------------------------------------------------------------- */
 SCRIPT(getObjectConfig);
 
 params [["_object", "", ["", objNull]]];
 
-if (_object isEqualType objNull) then {
-    _object = typeOf _object;
+private _isObject = _object isEqualType objNull;
+private _result = configNull;
+if _isObject then {
+    _result = _object getVariable ["CBA_config", configNull];
 };
 
-private _result = configNull;
+// Cached value has been found
+if (isClass _result) exitWith {_result};
+
+// Convert object to classname to be found if needed
+private _className = _object;
+if _isObject then {
+    _className = typeOf _object;
+};
+
+// Invalid class
+if (count _className == 0) exitWith {_result};
```

Comment:
```suggestion
if (_className == "") exitWith {_result};
```

Reply:
I used ` if (_className isEqualTo "") exitWith {_result};` as it seems to have tiny performance advantage even in this case.
```
Result:
3.69373 ms

Cycles:
271/10000

Code:
for "_i" from 0 to 10000 do {"" isEqualTo ""};
```
vs
```
Result:
3.7594 ms

Cycles:
266/10000

Code:
for "_i" from 0 to 10000 do {"" == ""};
```
I did about 20 measurements and these are the best ones I obtained, with `==` consistently slower than `isEqualTo`.